From afb6eb73c5cf2f6713aa4420bde26c52c2b7e1e0 Mon Sep 17 00:00:00 2001
From: Alex Bee <knaerzche@gmail.com>
Date: Thu, 16 Jun 2022 13:15:09 +0200
Subject: [PATCH 78/92] arm64: dts: use correct PLL for vdec core

vdec core should use codec pll for proper operation, by default
it uses general pll (GPLL) - as all other clocks would
---
 arch/arm64/boot/dts/rockchip/rk3399.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index 9590ffb70519..ebb6a6d63400 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -1346,6 +1346,8 @@ vdec: video-codec@ff660000 {
 		clocks = <&cru ACLK_VDU>, <&cru HCLK_VDU>,
 			 <&cru SCLK_VDU_CA>, <&cru SCLK_VDU_CORE>;
 		clock-names = "axi", "ahb", "cabac", "core";
+		assigned-clocks = <&cru ACLK_VDU>;
+		assigned-clock-parents = <&cru PLL_CPLL>;
 		iommus = <&vdec_mmu>;
 		power-domains = <&power RK3399_PD_VDU>;
 		resets = <&cru SRST_H_VDU>, <&cru SRST_A_VDU>,
-- 
2.34.1

